src/foam/box/SessionServerBox.java

```diff
@@ -108,7 +108,7 @@ public void send(Message msg) {
       logger.error("Error throw in SessionServerBox: " + t, " ,service: " + spec.getName(), t);
       t.printStackTrace();
       msg.replyWithException(t);
-      return;
+      throw t;
```

Comment:
@hchoura why are you doing this? The exception is added the 'msg' being returned. what good is rethrowing it as this point?